tmc-langs-util/src/main/java/fi/helsinki/cs/tmc/langs/util/tarservice/TarCreator.java

```diff
@@ -0,0 +1,99 @@
+package fi.helsinki.cs.tmc.langs.util.tarservice;
+
+import static org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.LONGFILE_POSIX;
+
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.utils.IOUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class TarCreator {
+
+    /**
+     * Copies the location of tmc-langs-cli and tmcrun to unzipped project folder
+     * and creates a tarball.
+     *
+     * @param projectDir     Location of unzipped project dir
+     * @param tmcLangs       Location of tmc-langs-cli.jar
+     * @param tmcrun         Location of tmc-run init script
+     * @param targetLocation Location where the tar archive should be extracted to
+     * @return Returns the location of the folder where the tarball was created
+     * @throws IOException      Error!
+     * @throws ArchiveException Error!
+     */
+
+    public Path createTarFromProject(Path projectDir, Path tmcLangs, Path tmcrun,
+                                     Path targetLocation) throws IOException, ArchiveException {
+        Files.copy(tmcrun, projectDir.resolve(tmcrun.getFileName().toString()));
+        Files.copy(tmcLangs, projectDir.resolve(tmcLangs.getFileName().toString()));
+        createTarBall(projectDir.toFile(), targetLocation);
+        return projectDir.getParent();
+    }
+
+    /**
+     * Creates a tarball from a directory.
+     *
+     * @param project        Project directory file
+     * @param targetLocation Location where the tar archive should be extracted to
+     * @throws IOException      Error!
+     * @throws ArchiveException Error!
+     */
+
+    private void createTarBall(File project, Path targetLocation)
+            throws IOException, ArchiveException {
+        FileOutputStream tarOut;
+        if (targetLocation.getFileName().toString().contains(".tar")) {
+            tarOut = new FileOutputStream(targetLocation.toString());
+        } else {
+            tarOut = new FileOutputStream(targetLocation.toString() + ".tar");
```

Comment:
Will anyone ever call this function with a `targetLocation` that doesn't end with `.tar`?